src/unit_tests/shared/test_json_op.c

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <errno.h>
+#include <string.h>
+#include <unistd.h>
+
+
+#include "../../headers/json_op.h"
+#include "../wrappers/externals/cJSON/cJSON_wrappers.h"
+#include "../wrappers/wazuh/shared/file_op_wrappers.h"
+#include "../wrappers/wazuh/shared/debug_op_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+#include "../wrappers/common.h"
```

Comment:
Insted of traversing directories you could modify the CMakeLists.txt to include those directories.

[CMakeLists example ](https://github.com/wazuh/wazuh/blob/fbb5ab9a11406c5e62437a2952ecfb36abe65398/src/shared_modules/rsync/tests/interface/CMakeLists.txt) for this [test file](https://github.com/wazuh/wazuh/blob/fbb5ab9a11406c5e62437a2952ecfb36abe65398/src/shared_modules/rsync/tests/interface/rsync_test.cpp)